### PR TITLE
Proposal: Static and class subscripts

### PR DESCRIPTION
**Pull request by brentdax**
The implementation is not quite completely tested yet, but it's far enough along that I think we can prepare the proposal for review.

### CODE REVIEW

**Comment by tkremenek** [comment]
@brentdax what are the current limitations of the implementation?

**Comment by brentdax** [comment]
@tkremenek The last open question that would affect the proposal's drafting is whether to keep metatype key paths in its scope. I'm currently working on emitting the metadata needed to support them in public resilient modules; I think I'll know by end of day tomorrow whether I want to include or defer them.

Besides that:

* I haven't tested existential metatypes very much yet, and I wouldn't be surprised if some of the current implementation only works for concrete metatypes.
* I haven't started seriously working on code completion or other editor features.

**Comment by brentdax** [comment]
I think there's enough doubt about resilient cross-module key paths that they should be removed from the proposal. I'll probably write the proposal revisions tonight, but certainly by lunch tomorrow.

File: proposals/0253-static-subscripts.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
+# Static and class subscripts
+
+* Proposal: [SE-0253](0253-static-subscripts.md)
```
Comment (krilnon) [inline proposals/0253-static-subscripts.md]: Usually it makes sense to delay numbering a proposal until it's merged. E.g. there's already an [SE-0253](https://github.com/apple/swift-evolution/blob/master/proposals/0253-callable.md).

Comment (brentdax) [reply proposals/0253-static-subscripts.md]: Yes, I'm quickly learning that lesson. 😅

Comment (DougGregor) [reply proposals/0253-static-subscripts.md]: No worries, I'll fix it

**Pull request merged**